### PR TITLE
add precommit rules for markdown/json files

### PR DESCRIPTION
**Pull request by joeyorlando**
**What this PR does**:

- Adds a `markdownlint-cli` `pre-commit` step. Uses most of the default rules, except for the few overrides in `.markdownlint.json` and `./docs/.markdownlint.json`. Decided to lint the `./docs` directory separately, with it's own set of overrides. The following two rules were turned off for `./docs`:
  - `MD025`:  Multiple top-level headings in the same document (because the nature of how metadata data is parsed at the top of these files)
  - `MD036`: - Emphasis used instead of a heading (for some emphasised elements we don't want to use a header so that it doesn't show up in the docs right hand navigation section)
```bash
docs/sources/oncall-api-reference/on_call_shifts.md:72 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/on_call_shifts.md:101 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/on_call_shifts.md:159 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/on_call_shifts.md:198 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/on_call_shifts.md:211 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/postmortem_messages.md:36 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/postmortem_messages.md:62 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/postmortem_messages.md:99 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/postmortem_messages.md:128 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
docs/sources/oncall-api-reference/postmortem_messages.md:140 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "HTTP request"]
```
- Adds a separate `prettier` `pre-commit` step just for `.json` files

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated (N/A)
- [ ] Documentation added (N/A)
- [ ] `CHANGELOG.md` updated (N/A)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,16 +17,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Errors and warnings that occur when rendering templates during notification or webhooks will now render
+  and display the error/warning as the result.
 
 ## v1.1.5 (2022-11-24)
 
 ### Added
 
-- Added a QR code in the "Mobile App Verification" tab on the user settings modal to connect the mobile application to your OnCall instance
-
-## v1.1.5 (2022-11-24)
```
Comment (joeyorlando) [inline CHANGELOG.md:29]: we had a duplicate `v1.1.5 (2022-11-24)` tag..

**Comment by joeyorlando** [comment]
majority of the changes were autofixed.
85% of the remaining errors were because line length was > 120 characters.

**Comment by mderynck** [comment]
Check with docs to make sure this doesn't conflict with other formatting. @alyssawada

**Review by joeyorlando — COMMENTED** [review]
.

**Review by alyssawada — APPROVED** [review]
This should be all good from a docs perspective! If there are additional rules that we'd like to override for any reason, I'll follow up.

**Review by vadimkerr — APPROVED** [review]

**Pull request merged**